### DISCUSSION TITLE
String helpers

### DISCUSSION THREAD

**Discussion by reindert-vetter** [confetti-framework/confetti#102] [Ideas]
> #97  @rafaelbreno @NWBY
One of the best helpers that I often use is the [tap()](https://laravel.com/docs/8.x/helpers#method-tap) functions, I've seen some approaches in Go, [here's](https://play.golang.org/p/WOnE31ldpb) one of them.

Example:
``` go
result := Wrap(input).
    Map(strings.ToUpper).
    Map(func(s string) string {
            return "prefix-" + s
	}).
    Tap(func(s string) {
            fmt.Printf("tap: %s\n", s)
	}).
    Filter(func(s string) bool {
            return len(s) >= 9
	}).
    Reduce(
        func(acc map[string]int, s string) map[string]int {
            acc[s] = len(s)
            return acc
	},
	map[string]int{}
    ).
    Val()
```

**Comment by reindert-vetter** [comment]
:construction: Due to the great success of Confetti, I don't have time to implement this. For the people who want to implement this it is best to create their own repostory. Then we can refer to it from the documentation.
If the repostory has 100% test coverage and contains more than 50% of the Laravel features, then we can (probably) recognize it as an official repo.

**Reply by Wulfheart** [reply]
Ok. I’m tempted to implement it next weeks after my tests.

**Comment by bosix** [comment]
Here is a full list of all helper functions laravel has: https://laravel.com/docs/8.x/helpers
